File: exercism/kotlin/difference-of-squares/src/main/kotlin/difference_of_squares.kt
class Squares(val input: Int) {

    fun squareOfSum(): Int {
        return (input downTo 1).fold(0) { sum, element -> sum + element } * (input downTo 1).fold(0) { sum, element -> sum + element }
    }

    fun sumOfSquares(): Int {
        return (input downTo 1).fold(0) { sum, element -> sum + element*element }
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}

File: exercism/ruby/difference-of-squares/difference_of_squares.rb
class Squares
   def initialize(numbers)
      @numbers = (1..numbers)
   end

   def sum_of_squares
       @numbers.map {|x| x*x}.sum
   end

   def square_of_sum
       @numbers.sum ** 2
   end

   def difference
       square_of_sum - sum_of_squares
   end
end

module BookKeeping
   VERSION = 4 
end
